### ISSUE TITLE
App sometimes loads with broken aspect

### ISSUE THREAD

**Issue by andydotxyz** [fyne-io/fyne#938]
Looks like some bad size calculations or resize.

The demo window sometimes loads too tall. This time it is the right size but with the wrong size of content.
Is this what you saw on WASM @Bluebugs?

OS: macOS 10.15.4
Fyne: develop

<img width="790" alt="Screenshot 2020-05-05 at 16 51 36" src="https://user-images.githubusercontent.com/294436/81086864-f3b14e00-8ef0-11ea-9f42-a6407ef9e178.png">

**Comment by andydotxyz** [comment]
Mouse events occur at the correct place but content refreshes at the wrong aspect, maintaining the squashed look until it is resized...

**Comment by andydotxyz** [comment]
This may be related - sometimes the window draws with incomplete content. That responds and refreshes in the same wrong look.
Only fixed by resizing.

<img width="790" alt="Screenshot 2020-05-05 at 16 55 25" src="https://user-images.githubusercontent.com/294436/81087187-50ad0400-8ef1-11ea-9504-ee64a71724e2.png">

**Comment by Bluebugs** [comment]
Indeed, it looks like exactly my problem. I have been slowly working on a fix as I am a bit busy this week, but basically it comes down to the idea that when Fyne ask the system to resize the window Fyne expect it to be always successful and instantaneous. Which is incorrect. Resizing the window is asynchronous in nature and it can be denied by the window manager, or the window manager can decide something else entirely.

My current solution is to only allow the window size change to happen on callback from glfw, not on request from Fyne. Fyne has to request glfw to resize the window and assume it didn't happen, fit the content in its current window size and maybe later have a better layout opportunity if the windows get resized.

**Comment by toaster** [comment]
Have seen this on MacOS 10.14, too.

**Comment by andydotxyz** [comment]
I think the underlying issue may be to do with changing minsize or something similar... It feels more like ordering of events or methods not returning consistent values than a window resize issue, but it needs more investigation.

This is a blocker on release as it's new on `develop`.

**Comment by Bluebugs** [comment]
I think you can reproduce the problem 100% of the time with a window that start fullscreen (before any content is set), and the content set is to big to fit in the screen. It will resize the gl viewport, but the window itself won't get resized and it will lead to exactly the same result. Another scenario where it would fail, would be in a tile window manager. Their is likely more case where the asynchronous nature of resize get in the way of the current logic and result in such artifact. I will try to do a patch this evening for it on top of `develop` instead on top of my wasm branch.

**Comment by andydotxyz** [comment]
A bigger problem is that you can get this every time if you choose the graphics tab in fyne_demo, exit then start it again. macOS >= 10.14

**Comment by stuartmscott** [comment]
Running the demo with:
```
go run -tags no_native_menus fyne.io/fyne/cmd/fyne_demo/main.go
```
causes menu to no render full width

<img width="679" alt="Screen Shot 2020-05-08 at 7 25 09 AM" src="https://user-images.githubusercontent.com/1117080/81415393-3fd0de00-90fd-11ea-8dfb-7612ce296dc6.png">

**Comment by andydotxyz** [comment]
@stuartmscott do you see the same broken menu on latest `develop`? It doesn't happen here any more.

**Comment by andydotxyz** [comment]
This plus other things seem to be fixed on my #966 PR as it addresses the ordering of some events :)

**Comment by andydotxyz** [comment]
On develop for testing :)
